mule-deployer/src/main/java/org/mule/tools/deployment/AbstractDeployerFactory.java

```diff
@@ -94,7 +94,7 @@ public Deployer createArtifactDeployer(Deployment deployment, DeployerLog log) t
         case MULE_CLASSIFIER:
           return new ArmApplicationDeployer(deployment, log);
         default:
-          throw new RuntimeException("Deployment not supported: " + deployment.getClass().getSimpleName());
+          throw new RuntimeException("Packaging not supported: " + deployment.getPackaging());
```

Comment:
I would rephrase it to something like this:
Packaging deployment.getPackaging() is not supported for the defined deployment target.